cdm/gcdm/src/test/java/ucar/gcdm/TestGcdmNetcdfFileProblem.java

```diff
@@ -0,0 +1,371 @@
+      assertThat(size).isGreaterThan(GcdmServer.MAX_MESSAGE); // Data should be chunked
+
+      ArrayStructure arrayStructure = (ArrayStructure) variable.read();
+      Array memberData = arrayStructure.extractMemberArray(arrayStructure.findMember(memberName));
+      assertThat(memberData.getSize()).isEqualTo(expectedData.getSize());
+      for (int i = 0; i < memberData.getSize(); i++) {
+        assertThat(memberData.getInt(i)).isEqualTo(expectedData.getInt(i));
+      }
+    }
+  }
+
+  @Test
+  @Category(NeedsCdmUnitTest.class)
+  public void testSequenceData() throws IOException {
+    String filename = TestDir.cdmUnitTestDir + "formats/bufr/userExamples/test1.bufr";
+    String gcdmUrl = gcdmPrefix + filename;
+
+    try (GcdmNetcdfFile gcdmFile = GcdmNetcdfFile.builder().setRemoteURI(gcdmUrl).build()) {
+      Sequence sequence = (Sequence) gcdmFile.findVariable("obs");
+      assertThat((Object) sequence).isNotNull();
+      assertThat(sequence.getNumberOfMemberVariables()).isEqualTo(78);
+
+      Variable variable = sequence.findVariable("Station_or_site_name");
+      assertThat((Object) variable).isNotNull();
+      assertThat(variable.getSize()).isEqualTo(20);
+      assertThat(variable.getDataType()).isEqualTo(DataType.CHAR);
+
+      ArrayChar data = (ArrayChar) variable.read();
+      assertThat(data.getString(0)).isEqualTo("WOLLOGORANG        ");
+    }
+  }
+
+  private static void compareArrayToArray(Path path) throws Exception {
+    String gcdmUrl = gcdmPrefix + path.toAbsolutePath();
+    try (NetcdfFile ncfile = NetcdfDatasets.openFile(path.toString(), null);
+        GcdmNetcdfFile gcdmFile = GcdmNetcdfFile.builder().setRemoteURI(gcdmUrl).build()) {
+
+      Formatter formatter = new Formatter();
+      boolean ok = CompareNetcdf2.compareFiles(ncfile, gcdmFile, formatter, true, false, true);
+      logger.debug(formatter.toString());
+      assertThat(ok).isTrue();
+    }
+  }
+
+  private static void compareArrayToArray(Path path, String varName) throws Exception {
+    String gcdmUrl = gcdmPrefix + path.toAbsolutePath();
+    try (NetcdfFile ncfile = NetcdfDatasets.openFile(path.toString(), null);
+        GcdmNetcdfFile gcdmFile = GcdmNetcdfFile.builder().setRemoteURI(gcdmUrl).build()) {
+
+      Variable ncVar = ncfile.findVariable(varName);
+      Variable gcdmVar = gcdmFile.findVariable(varName);
+
+      Formatter formatter = new Formatter();
+      CompareNetcdf2 compareNetcdf2 = new CompareNetcdf2(formatter, false, true, true);
+      boolean ok = compareNetcdf2.compareVariable(ncVar, gcdmVar, CompareNetcdf2.IDENTITY_FILTER);
+      logger.debug(formatter.toString());
+      assertThat(ok).isTrue();
+    }
+  }
+
+  private static void writeNetcdfFile(String filename, String variableName, DataType dataType, Array values)
+      throws IOException, InvalidRangeException {
+    NetcdfFormatWriter.Builder writerBuilder = NetcdfFormatWriter.createNewNetcdf3(filename);
```

Comment:
Can this be refactored so that instead of passing filename in, the tempfolder creates a file in this method? As an extra safety catch against making persistent files in tests.

Reply:
Yes, good idea-- fixed in 21acb5d93b